File: cs-budget-api/main/src/Models/ProcessedTransaction.cs
using System.Text.Json.Serialization;
using Utils;

namespace Models;

public class ProcessedTransaction(Guid id, string category, string description, decimal value, DateTimeOffset timestamp)
{
    public Guid Id { get; set; } = id;
    public string Category { get; set; } = category;
    public string Description { get; set; } = description;
    public decimal Value { get; set; } = value;

    [JsonConverter(typeof(DateTimeOffsetConverter))]
    public DateTimeOffset Timestamp { get; set; } = timestamp;
}

File: go-budget-api/internal/models/processed_transaction.go
package models

import (
	"encoding/json"
	"time"
)

type ProcessedTransaction struct {
	Id          string    `json:"id"`
	Category    string    `json:"category"`
	Description string    `json:"description"`
	Value       float64   `json:"value"`
	Timestamp   time.Time `json:"timestamp"`
}

func (u *ProcessedTransaction) MarshalJSON() ([]byte, error) {
	return json.Marshal(&struct {
		Id          string  `json:"id"`
		Category    string  `json:"category"`
		Description string  `json:"description"`
		Value       float64 `json:"value"`
		Timestamp   string  `json:"timestamp"`
	}{
		Id:          u.Id,
		Category:    u.Category,
		Description: u.Description,
		Value:       u.Value,
		Timestamp:   u.Timestamp.Format("2006-01-02T15:04:05.000Z"),
	})
}
